include/seqan3/alignment/pairwise/align_result.hpp

```diff
@@ -34,210 +34,201 @@
 
 /*!\file
  * \brief Provides seqan3::align_result.
+ * \author Jörg Winkler <j.winkler AT fu-berlin.de>
  * \author Rene Rahn <rene.rahn AT fu-berlin.de>
  */
 
 #pragma once
 
-#include <functional>
-#include <optional>
-#include <tuple>
-
 #include <seqan3/core/metafunction/template_inspection.hpp>
 
-namespace seqan3
+namespace seqan3::detail
 {
 
-/*!\brief Keys for different alignment results.
- * \ingroup pairwise
+/*!\brief A struct that contains the actual alignment result data.
+ * \tparam id_t          The type for the alignment identifier.
+ * \tparam score_t       The type for the resulting score.
+ * \tparam end_coord_t   The type for the end coordinate, can be omitted.
+ * \tparam begin_coord_t The type for the begin coordinate, can be omitted.
+ * \tparam trace_t       The type for the alignment trace, can be omitted.
```

Comment:
Can you please rename trace_t and all of its variants to alignment_t. Since we call the result an alignment.